### PR TITLE
mon.te patches as well as some fstools patches related to it

### PR DESCRIPTION
**Pull request by etbe**

### CODE REVIEW

File: policy/modules/kernel/files.if [hunk 1]
```diff
@@ -434,6 +434,24 @@ interface(`files_tmpfs_file',`
...
+## <summary>
+##	dontaudit getattr on tmpfs files
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain to not have stat on tmpfs files audited
+##	</summary>
+## </param>
+#
+interface(`files_dontaudit_tmpfs_file_getattr',`
```
Comment (pebenito) [inline policy/modules/kernel/files.if:447]: Please rename to `files_dontaudi_getattr_all_tmpfs_files`.

File: policy/modules/kernel/storage.if [hunk 2]
```diff
@@ -310,6 +310,30 @@ interface(`storage_dev_filetrans_fixed_disk',`
...
+##	Domain allowed access.
+##	</summary>
+## </param>
+## <param name="filename" optional="true">
+##	<summary>
+##	Optional filename of the char device to be created
+##	</summary>
+## </param>
+#
+interface(`storage_dev_filetrans_fixed_disk_control',`
```
Comment (pebenito) [inline policy/modules/kernel/storage.if:329]: Please remove `_control` from the name and make the obj class a parameter.

File: policy/modules/services/mon.te [hunk 3]
```diff
@@ -161,9 +169,10 @@ optional_policy(`
 #
 
 # sys_ptrace is for reading /proc/1/maps etc
-allow mon_local_test_t self:capability { sys_ptrace sys_admin };
+allow mon_local_test_t self:capability { dac_override dac_read_search setgid setuid sys_rawio sys_ptrace sys_admin };
```
Comment (pebenito) [inline policy/modules/services/mon.te:172]: The `storage_raw_read_fixed_disk` should have the `sys_rawio` with it, so I'd rather keep it there and drop it here, unless there is another reason to need the perm.

File: policy/modules/system/fstools.te [hunk 4]
```diff
@@ -145,6 +150,8 @@ mls_file_write_all_levels(fsadm_t)
 
 selinux_getattr_fs(fsadm_t)
 
+storage_dev_filetrans_fixed_disk_control(fsadm_t, "megaraid_sas_ioctl_node")
```
Comment (pebenito) [inline policy/modules/system/fstools.te:153]: I'd rather drop the name unless there is a conflicting type_transition that I'm not seeing.

**Review by pebenito — APPROVED** [review]

**Review by pebenito — CHANGES_REQUESTED** [review]
accidentally hit approve on above comments instead of request changes.

File: policy/modules/kernel/files.if (hunk 1 above)
Comment (etbe) [reply policy/modules/kernel/files.if]: done

File: policy/modules/kernel/storage.if (hunk 2 above)
Comment (etbe) [reply policy/modules/kernel/storage.if]: There was already an interface without the _control. Following the example of dev_filetrans_cardmgr() I made it take both chr_file and blk_file,

File: policy/modules/services/mon.te (hunk 3 above)
Comment (etbe) [reply policy/modules/services/mon.te]: done

File: policy/modules/system/fstools.te (hunk 4 above)
Comment (etbe) [reply policy/modules/system/fstools.te]: ok

Comment (etbe) [reply policy/modules/system/fstools.te]: libsepol.expand_terule_helper: conflicting TE rule for (kernel_t, device_t:chr_file):  old was event_device_t, new is fixed_disk_device_t
libsepol.expand_module: Error during expand

Seems to be a conflicting transition

Comment (etbe) [reply policy/modules/system/fstools.te]: dev_filetrans_input_dev(kernel_t)
storage_dev_filetrans_fixed_disk(kernel_t)

The existence of the above 2 lines precludes having storage_dev_filetrans_fixed_disk() work on both chr_file and blk_file. Shall I replace storage_dev_filetrans_fixed_disk with 2 new interfaces and if so what shall I name them?

Comment (pebenito) [reply policy/modules/system/fstools.te]: In that case, I propose: try adding `chr_file` to `storage_dev_filetrans_fixed_disk()` and then keep the `"megaraid_sas_ioctl_node"` in the call.

Comment (etbe) [reply policy/modules/system/fstools.te]: ok

**Pull request merged**